documentation/boltspec_reference.md

```diff
@@ -1225,7 +1227,562 @@ The `allow_upload` function accepts the following modifiers:
+    results = targets.map do |target|
+      Bolt::Result.new(target, value: { 'path' => File.join(destination, source) })
+    end
+
+    Bolt::ResultSet.new(results)
+  end
+  ```
+
+- `error_with(error)`
+
+  Sets the error hash for each target's `Result` when the file is downloaded.
+  Returns a `Bolt::ResultSet` object.
+
+  ```ruby
+  allow_any_download.error_with('msg' => 'File not found')
+
+### `allow_any_out_message`
+
+The `allow_any_out_message` function stubs all invocations of the [`out::message`
+function](plan_functions.md#out-message). It does not accept parameters.
+
+The `allow_any_out_message` function accepts the following modifiers:
+
+- `be_called_times(number)`
+
+  The test fails if `out::message` is run more than _number_ times. 
+
+  ```ruby
+  allow_any_out_message.be_called_times(3)
+  ```
+
+- `not_be_called`
+
+  The test fails if `out::message` is called.
+
+  ```ruby
+  allow_any_out_message.not_be_called
+  ```
+
+- `with_params(parameters)`
+
+  The parameters that can be passed to the `out::message` function. 
+  The test fails if `out::mesage` is called with a different
+  set of parameters.
+
+  ```ruby
+  allow_any_out_message.with_params("hello world")
+  ```
+
+### `allow_any_plan`
+
+The `allow_any_plan` function stubs all invocations of the [`run_plan`
+function](plan_functions.md#run-plan). It does not accept parameters.
+
+The `allow_any_plan` function accepts the following modifiers:
+
+- `be_called_times(number)`
+
+  The test fails if any plan is run more than _number_ times. 
+
+  ```ruby
+  allow_any_plan.be_called_times(3)
+  ```
+
+- `not_be_called`
+
+  The test fails if any plan is run.
+
+  ```ruby
+  allow_any_plan.not_be_called
+  ```
+
+- `with_params(parameters)`
+
+  The parameters and [options](plan_functions.md#run-plan) that can be passed to
+  the `run_plan` function. The test fails if any plan is run with a different
+  set of parameters and options.
+
+  ```ruby
+  allow_any_plan.with_params({ 'fruit' => 'apple', '_run_as' => 'root' })
+  ```
+
+- `always_return(value)`
+
+  Sets the value for the `PlanResult` object returned by every plan. The
+  `PlanResult` object returned by this modifier always has a `success` status.
+
+  ```ruby
+  allow_any_plan.always_return(318)
+  ```
+
+- `return(&block) { |plan, params| ... }`
+
+  Invokes a block to construct the `PlanResult` returned by the `run_plan`
+  function.
+
+  ```ruby
+  allow_any_plan.return do |plan, params|
+    Bolt::PlanResult.new(100, 'success')
+  end
+  ```
+
+- `error_with(error)`
+
+  Sets the value for the `PlanResult` object returned by each plan. The
+  `PlanResult` object returned by this modifier always has a `failure` status.
+
+  ```ruby
+  allow_any_plan.error_with('Too many apples, buffer overflow!')
+  ```
+
+### `allow_any_script`
+
+The `allow_script` function stubs all invocations of the [`run_script`
+function](plan_functions.md#run-script). It does not accept parameters.
+
+```ruby
+allow_script
```

Comment:
```suggestion
allow_any_script
```

This code block is missing under `allow_any_out_message`, `allow_any_out_verbose`, `allow_any_task`, and `allow_any_plan`.